multiapps-controller-web/src/test/java/org/cloudfoundry/multiapps/controller/web/upload/AsyncUploadJobExecutorTest.java

```diff
@@ -0,0 +1,824 @@
+
+        when(asyncUploadJobsQuery.singleResult())
+            .thenReturn(runningEntry)
+            .thenReturn(finishedEntry);
+
+        setupSuccessfulHttpResponseMocks();
+
+        when(fileService.addFile(any(FileEntry.class), any(InputStream.class))).thenReturn(fileEntry);
+        when(fileService.processFileContent(anyString(), anyString(), any())).thenReturn(deploymentDescriptor);
+
+        when(descriptorParserFacade.parseDeploymentDescriptor(anyString())).thenReturn(deploymentDescriptor);
+
+        try (MockedStatic<SecurityContextUtil> mockedSecurityContext = mockStatic(SecurityContextUtil.class)) {
+            mockedSecurityContext.when(SecurityContextUtil::getUsername)
+                                 .thenReturn(USERNAME);
+
+            AsyncUploadJobEntry result = asyncUploadJobExecutor.executeUploadFromUrl(
+                SPACE_GUID, NAMESPACE, FILE_URL, DECODED_URL, userCredentials);
+
+            assertNotNull(result);
+            assertEquals(SPACE_GUID, result.getSpaceGuid());
+            assertEquals(NAMESPACE, result.getNamespace());
+            assertEquals(FILE_URL, result.getUrl());
+            assertEquals(AsyncUploadJobEntry.State.INITIAL, result.getState());
+
+            verify(asyncUploadJobService).add(any(AsyncUploadJobEntry.class));
+            verify(deployFromUrlExecutor).submit(any(Runnable.class));
+        }
+    }
+
+    @Test
+    void testInsecureUrlThrowsException() {
+        AsyncUploadJobEntry initialEntry = createInitialJobEntry();
+        AsyncUploadJobEntry runningEntry = ImmutableAsyncUploadJobEntry.copyOf(initialEntry)
+                                                                       .withState(AsyncUploadJobEntry.State.RUNNING)
+                                                                       .withStartedAt(LocalDateTime.now());
+        AsyncUploadJobEntry errorEntry = ImmutableAsyncUploadJobEntry.copyOf(runningEntry)
+                                                                     .withState(AsyncUploadJobEntry.State.ERROR)
+                                                                     .withError(Messages.MTAR_ENDPOINT_NOT_SECURE);
+
+        when(asyncUploadJobService.add(any(AsyncUploadJobEntry.class))).thenReturn(initialEntry);
+        when(asyncUploadJobService.update(any(AsyncUploadJobEntry.class), any(AsyncUploadJobEntry.class)))
+            .thenReturn(runningEntry)
+            .thenReturn(errorEntry);
+        when(asyncUploadJobsQuery.singleResult()).thenReturn(runningEntry);
+
+        try (MockedStatic<SecurityContextUtil> mockedSecurityContext = mockStatic(SecurityContextUtil.class)) {
+            mockedSecurityContext.when(SecurityContextUtil::getUsername)
+                                 .thenReturn(USERNAME);
+
+            AsyncUploadJobEntry result = asyncUploadJobExecutor.executeUploadFromUrl(
+                SPACE_GUID, NAMESPACE, INSECURE_URL, INSECURE_URL, userCredentials);
+
+            assertNotNull(result);
```

Comment:
Can we replace this type of assert in all test cases with assertEquals?

Reply:
done